### DISCUSSION TITLE
v20.1.1

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3494] [Releases]
After the colossal 20.1.0 release just a few weeks back, it's time for a little "smooth the edges" update, but don't be fooled, it's as important as your morning coffee!

## Highlights of OpenMage 20.1.1
* 🛠️ ZF1Future version in composer is strictly specified now, it order to fix problems with future upstream upgrades and our composer patches
* 🚀 Totally **new JSON-RPC API is now available**, with support for **instal-login via HTTP basic auth**, it seems to be much more performant than the SOAP one so definitely check it out
* 📦 **USPS module got a nice upgrade** to support the latest changes in USPS APIs and services
* 🍪 A **vulnerability about "guest order view" cookie tampering was fixed**, with the introduction of a **"rate limiting" API** you can use in your own code too
* 🐞 Few other bugfix and updates

## Complete changelog
* Bump phpstan/phpstan from 1.10.26 to 1.10.27 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3434
* Allowed admin grid row to open in new tab by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/3425
* Upgraded ZF1Future 1.22.1, Credis 1.15 and some minor dependencies by @fballiano in https://github.com/OpenMage/magento-lts/pull/3437
* Bump shardj/zf1-future from 1.22.1 to 1.23.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3442
* Updated countries list by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3436
* Added JSON-RPC API support by @fballiano in https://github.com/OpenMage/magento-lts/pull/2273
* Allowed API insta-login via HTTP Basic Auth by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3443
* More verbose PHP errors logging for API by @kyrena in https://github.com/OpenMage/magento-lts/pull/3427
* Added info about v21 to the README by @fballiano in https://github.com/OpenMage/magento-lts/pull/3388
* Updated zf1f 1.23.2, phpstan 1.10.29 etc by @fballiano in https://github.com/OpenMage/magento-lts/pull/3446
* Bump phpunit/phpunit from 9.6.10 to 9.6.11 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3456
* Fixing invalid null-return in Google Analytics  by @daboss84 in https://github.com/OpenMage/magento-lts/pull/3438
* Allowed soap_v2 style request in JSON-RPC and added event 'api_server_adpter_jsonrpc_run_after' by @kiatng in https://github.com/OpenMage/magento-lts/pull/3444
* Fixed conflict between Prototype.js and Google (#3459) by @Tomasz-Silpion in https://github.com/OpenMage/magento-lts/pull/3460
* Bump tj-actions/changed-files from 37 to 38 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3471
* Bump shardj/zf1-future from 1.23.2 to 1.23.5 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3470
* Bump phpstan/phpstan from 1.10.29 to 1.10.32 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3469
* Mage_Usa: added "USPS Ground Advantage" and removed "First Class Package" by @fballiano in https://github.com/OpenMage/magento-lts/pull/3413
* Fixed out-of-range column `lognum` in the table `api_user` by @kiatng in https://github.com/OpenMage/magento-lts/pull/3480
* Fixed ZF1F version in composer.json to avoid conflicts with our patches by @fballiano in https://github.com/OpenMage/magento-lts/pull/3475
* Bump symfony/polyfill-php74 from 1.27.0 to 1.28.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3487
* Bump friendsofphp/php-cs-fixer from 3.23.0 to 3.25.1 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3490
* Fixed a Type Error when converting an Order with no Coupon to a Quote by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3492
* Added composer patch to get.php, ref. #3453 by @fballiano in https://github.com/OpenMage/magento-lts/pull/3476
* Fixed unnecessary entries in table `api-session` when using insta-login in API calls by @kiatng in https://github.com/OpenMage/magento-lts/pull/3477

## New Contributors
* @daboss84 made their first contribution in https://github.com/OpenMage/magento-lts/pull/3438

**Full Changelog**: https://github.com/OpenMage/magento-lts/compare/v20.1.0...v20.1.1

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v20.1.1'>v20.1.1</a>.</em>

**Comment by m-overlund** [comment]
Upgrading on non-composer install:
- is it enough to download a patch file from github? (upgrading from 20.1.0)
- or do you have to download the package file in order to get composer related updates, like ZF future changes etc.?

**Reply by fballiano** [reply]
you have to run composer install or use the patch file + the vendor folder from the zip

**Reply by m-overlund** [reply]
of course - didn't think it through - forgot i could just update vendors via Composer even though OpenMage is not installed via Composer.
Thank you
